### PR TITLE
7726 Replace Serilog from Codebase and implement Open Telemetry

### PR DESCRIPTION
**Pull request by alexanderwjrussell**
#### 📲 What

Replace Serilog from Codebase and implement Open Telemetry

#### 🤔 Why

OpenTelemetry is lighter weight and supports all the functionality we require.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs [hunk 1]
```diff
@@ -34,34 +35,54 @@
...
-                });
-                builder.AddOtlpExporter(options =>
-                {
-                    options.Endpoint = new Uri(Environment.GetEnvironmentVariable(Constants.EnvironmentVariables.OltpEndpoint));
-                });
-            });
-}
+AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+builder.Services.AddOpenTelemetry()
+    .WithTracing(tracingBuilder =>
```
Comment (sharonrussell) [inline src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs:47]: Can these be chained?

**Comment by alexanderwjrussell** [comment]
Example logs from the console:

image

File: src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs [hunk 2]
```diff
@@ -34,34 +35,54 @@
...
+    .WithLogging(loggerProviderBuilder =>
+    {
+        loggerProviderBuilder.ConfigureResource(resource =>
+        {
+            resource.AddService(otlpServiceName);
+        });
+        loggerProviderBuilder.AddConsoleExporter();
+    });
+
+builder.Services.AddOpenTelemetry()
```
Comment (sharonrussell) [inline src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs:81]: And this

File: src/func-aeh-listener/src/functions/xxENSONOxx.xxSTACKSxx.Listener/Program.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                loggingBuilder.AddOpenTelemetry(options =>
+                {
+                    options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("xxENSONOxx.xxSTACKSxx.Listener"));
+                    options.IncludeFormattedMessage = true;
+                    options.IncludeScopes = true;
+                    options.AddConsoleExporter();
+                });
+            });
+            
+        services.AddOpenTelemetry().WithTracing(builder =>
```
Comment (sharonrussell) [inline src/func-aeh-listener/src/functions/xxENSONOxx.xxSTACKSxx.Listener/Program.cs:37]: Same re: chaining

File: src/func-asb-listener/src/functions/xxENSONOxx.xxSTACKSxx.Listener/Program.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                loggingBuilder.AddOpenTelemetry(options =>
+                {
+                    options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("xxENSONOxx.xxSTACKSxx.Listener"));
+                    options.IncludeFormattedMessage = true;
+                    options.IncludeScopes = true;
+                    options.AddConsoleExporter();
+                });
+            });
+            
+        services.AddOpenTelemetry().WithTracing(builder =>
```
Comment (sharonrussell) [inline src/func-asb-listener/src/functions/xxENSONOxx.xxSTACKSxx.Listener/Program.cs:37]: chaining

File: src/func-cosmosdb-worker/src/functions/xxENSONOxx.xxSTACKSxx.Worker/Program.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                loggingBuilder.AddOpenTelemetry(options =>
+                {
+                    options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("xxENSONOxx.xxSTACKSxx.Listener"));
+                    options.IncludeFormattedMessage = true;
+                    options.IncludeScopes = true;
+                    options.AddConsoleExporter();
+                });
+            });
+            
+        services.AddOpenTelemetry().WithTracing(builder =>
```
Comment (sharonrussell) [inline src/func-cosmosdb-worker/src/functions/xxENSONOxx.xxSTACKSxx.Worker/Program.cs:37]: chaining

File: src/simple-api/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs [hunk 6]
```diff
@@ -43,27 +40,47 @@
...
+            resource.AddService(otlpServiceName);
         });
-}
+        tracingBuilder.AddAspNetCoreInstrumentation();
+        tracingBuilder.AddConsoleExporter(options =>
+        {
+            options.Targets = ConsoleExporterOutputTargets.Debug;
+        });
+    });
+builder.Services.AddOpenTelemetry()
```
Comment (sharonrussell) [inline src/simple-api/src/api/xxENSONOxx.xxSTACKSxx.API/Program.cs:57]: More chaining

**Review by sharonrussell — CHANGES_REQUESTED** [review]
Very minor comments

**Review by sharonrussell — APPROVED** [review]

**Pull request merged**